lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -2135,6 +2139,8 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
                 (`shouldBe` Incoming)
             , expectField (#amount . #getQuantity)
                 (`shouldBe` (oneMillionAda - fee))
+            , expectField (#fee . #getQuantity)
+                (between (estimatedFeeMin, estimatedFeeMax))
```

Comment:
Hi @Anviking 

Just curious: what's the reason for adding this assertion here (i.e., that the eventual fee never escapes the estimate bounds)?

Is there some way we could avoid writing an assertion like this (which propagates the mistaken idea that eventual fees should be bounded by earlier estimates), and instead add extra coverage of `balanceTx` in the case of external reward accounts?

Reply:
> Just curious: what's the reason for adding this assertion here (i.e., that the eventual fee never escapes the estimate bounds)?

https://input-output-rnd.slack.com/archives/C04T7AEDY4X/p1680693892720819?thread_ts=1680693183.017249&cid=C04T7AEDY4X

I changed the `fee` variable for the `amount` check to refer to the actual fee, rather than `estimatedFeeMax`. Then I added this `fee 'between' (estimatedFeeMin, estimatedFeeMax)` as a sanity-check.

Reply:
Maybe if we also add link to a task for removing the checks? 🤔